l1/l1.go

```diff
@@ -76,19 +76,25 @@ func (c *Client) subscribeToUpdates(ctx context.Context, updateChan chan *contra
 }
 
 func (c *Client) checkChainID(ctx context.Context) error {
-	gotChainID, err := c.l1.ChainID(ctx)
-	if err != nil {
-		return fmt.Errorf("retrieve Ethereum chain ID: %w", err)
-	}
+	for {
+		backOffDuration := time.Duration(1) * time.Second
```

Comment:
This should go outside the `for` loop, otherwise we reset the backoff every iteration.